tests/enterprise-contract/contract.go

```diff
@@ -0,0 +1,146 @@
+	"github.com/redhat-appstudio/e2e-tests/pkg/utils/tekton"
+)
+
+var _ = framework.EnterpriseContractSuiteDescribe("Enterprise Contract E2E tests", Label("enterprise-contract", "HACBS"), func() {
+
+	defer GinkgoRecover()
+	var fwk *framework.Framework
+	var err error
+	var namespace string
+	var kubeController tekton.KubeController
+	var policySource []ecp.Source
+	var imageWithDigest string
+	var pipelineRunTimeout int
+	var generator tekton.VerifyEnterpriseContract
+	var verifyECTaskBundle string
+	publicSecretName := "cosign-public-key"
+
+	BeforeAll(func() {
+		fwk, err = framework.NewFramework(constants.TEKTON_CHAINS_E2E_USER)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(fwk.UserNamespace).NotTo(BeNil(), "failed to create sandbox user")
+		namespace = fwk.UserNamespace
+		kubeController = tekton.KubeController{
+			Commonctrl: *fwk.AsKubeAdmin.CommonController,
+			Tektonctrl: *fwk.AsKubeAdmin.TektonController,
+			Namespace:  namespace,
+		}
+		publicKey, err := kubeController.GetTektonChainsPublicKey()
+		Expect(err).ToNot(HaveOccurred())
+		GinkgoWriter.Printf("Copy public key from %s/signing-secrets to a new secret\n", constants.TEKTON_CHAINS_NS)
+		Expect(kubeController.CreateOrUpdateSigningSecret(
+			publicKey, publicSecretName, namespace)).To(Succeed())
+
+		defaultEcp, err := kubeController.GetEnterpriseContractPolicy("default", "enterprise-contract-service")
+		Expect(err).NotTo(HaveOccurred())
+		policySource = defaultEcp.Spec.Sources
+
+		cm, err := kubeController.Commonctrl.GetConfigMap("ec-defaults", "enterprise-contract-service")
+		Expect(err).ToNot(HaveOccurred())
+		verifyECTaskBundle = cm.Data["verify_ec_task_bundle"]
+		Expect(verifyECTaskBundle).ToNot(BeEmpty())
+		GinkgoWriter.Printf("Using verify EC task bundle: %s\n", verifyECTaskBundle)
+
+	})
+
+	Context("Release Policy", func() {
+		BeforeAll(func() {
+			generator = tekton.VerifyEnterpriseContract{
+				Bundle:              verifyECTaskBundle,
+				Image:               imageWithDigest,
+				Name:                "verify-enterprise-contract",
+				Namespace:           namespace,
+				PolicyConfiguration: "ec-policy",
+				PublicKey:           fmt.Sprintf("k8s://%s/%s", namespace, publicSecretName),
+				SSLCertDir:          "/var/run/secrets/kubernetes.io/serviceaccount",
+				Strict:              false,
+			}
+			secretName := fmt.Sprintf("golden-image-public-key%s", util.GenerateRandomString(10))
+			//The staging public key for verificaiton image
+			publicKey := []byte("-----BEGIN PUBLIC KEY-----\n" +
+				"MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEZP/0htjhVt2y0ohjgtIIgICOtQtA\n" +
+				"naYJRuLprwIv6FDhZ5yFjYUEtsmoNcW7rx2KM6FOXGsCX3BNc7qhHELT+g==\n" +
+				"-----END PUBLIC KEY-----")
+
+			GinkgoWriter.Println("Create golden image public signing key")
+			Expect(kubeController.CreateOrUpdateSigningSecret(publicKey, secretName, namespace)).To(Succeed())
+			generator.PublicKey = fmt.Sprintf("k8s://%s/%s", namespace, secretName)
```

Comment:
You could handle the secret creation earlier so you can use it when `generator` is created.

Reply:
updated with codes. PTAL.